lib/rust/ensogl/core/src/display/scene.rs

```diff
@@ -740,6 +780,8 @@ impl HardcodedLayers {
             DETACHED,
             root,
             viz,
+            viz_resize_grip,
+            viz_overlay,
```

Comment:
If I understand EnsoGL correctly, we do not have any actual performance gains from using Rectangle if it forces us to create new layers, as the layers are a separate draw calls anyway. Perhaps the resize_grip could be a custom shape? That would make the code simpler (we already have a bit too many layers).

If you add new layers because you need to set ordering of the rectangles, you should probably use instance ordering implemented in https://github.com/enso-org/enso/pull/6140 instead.

Reply:
I'm not using layers here, but rather partition layers implemented in #6140, as you suggested. All these added layers are "fake" ones, so everything should be ok here.

Reply:
Ah, I forgot about partition layers (and that they are in the same place as normal layers).